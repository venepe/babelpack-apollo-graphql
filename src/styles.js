import { injectGlobal } from 'styled-components'


export const theme = {
  layout: {
    maxWidth: `${1000 / 16}rem`,
  },
  color: {
    primary: '#5d9cec',
    secondary: '#4a89dc',
    lightGreen: '#48cfad',
    darkGreen: '#37bc9b',
    lightRed: '#ed5565',
    darkRed: '#da4453',
    white: '#fff',
    lightGrey: '#ccd1d9',
    darkGrey: '#aab2bd',
    black: '#000',
  },
};

const breakpoints = {
  xs: 340,
  sm: 550,
  md: 768,
  lg: 1000,
  xl: 1200,
  xxl: 1600,
};

export const mediaQueries = Object.keys(breakpoints).reduce((obj, size) => {
  obj[size] = `@media (min-width: ${breakpoints[size] / 16}rem)`
  return obj
}, {});

const injectGlobalStyles = () => injectGlobal`
  html,
  body,
  #__next {
    height: 100%;
  }

  html *
{
   font-family: 'Cormorant Garamond' !important;
}

  body {
    min-width: 326px;
  }
  body,
  textarea,
  input,
  button,
  label {
    font-family: 'Cormorant Garamond', 'Open Sans', sans-serif;
  }
`

export default injectGlobalStyles;
