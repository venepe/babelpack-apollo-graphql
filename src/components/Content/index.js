import styled from 'styled-components'
import { mediaQueries } from 'styles'


const Content = styled.div`
  padding: 15px;
  display: flex;
  flex-direction: column;
  flex: 1;
  margin-top: 10px;

  ${mediaQueries.md} {
    padding: 25px 25px;
  }
`

export default Content
