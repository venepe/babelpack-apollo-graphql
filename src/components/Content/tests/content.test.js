import React from 'react'
import { shallow } from 'enzyme'
import { theme } from 'styles'
import 'jest-styled-components'
import Content from '../index'


describe('Content Component', () => {
  it('has correct default styles', () => {
    const wrapper = shallow(
      <Content theme={theme}>Some Content</Content>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
