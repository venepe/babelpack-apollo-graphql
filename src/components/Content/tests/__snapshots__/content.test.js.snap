// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component has correct default styles 1`] = `
.c0 {
  padding: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 10px;
}

@media (min-width:48rem) {
  .c0 {
    padding: 25px 25px;
    max-width: 62.5rem;
  }
}

<div
  className="c0"
>
  Some Content
</div>
`;
